src/backend/d3d12/RenderPipelineD3D12.cpp

```diff
@@ -120,12 +121,14 @@ namespace d3d12 {
         descriptor.BlendState.RenderTarget[0].LogicOp = D3D12_LOGIC_OP_NOOP;
         descriptor.BlendState.RenderTarget[0].RenderTargetWriteMask = D3D12_COLOR_WRITE_ENABLE_ALL;
 
-        descriptor.DepthStencilState.DepthEnable = false;
-        descriptor.DepthStencilState.StencilEnable = false;
+		DepthStencilState* depthStencilState = ToBackend(GetDepthStencilState());
+		descriptor.DepthStencilState = depthStencilState->GetD3D12DepthStencilDescriptor();
+
         descriptor.SampleMask = UINT_MAX;
         descriptor.PrimitiveTopologyType = D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE;
         descriptor.NumRenderTargets = 1;
         descriptor.RTVFormats[0] = DXGI_FORMAT_R8G8B8A8_UNORM;
+        descriptor.DSVFormat = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
```

Comment:
This change LGTM, the following can be addressed in a follow-up patch.

NumRenderTargets, RTVFormats and DSVFormats should depend on the information that has been set for the (renderpass, subpass) for this pipeline. Right now the Debug layer will complain in the many cases where we don't have a DSV set in OMSetRenderTargets. Also @kainino0x we should make sure we include some multiple attachment tests when testing render passes.